### ISSUE TITLE
Choosing the content of a tab stop from a list during completion

### ISSUE THREAD

**Issue by dahanbn** [garbas/vim-snipmate#240]
Is there a way to define multiple values for a tab stop and if I expand the snippet and reach that tab stop I can somehow select which value should used for that tab stop. 

After that the snippet should be expanded as usual (go to next tab stop). 

Can I get that behaviour with VimScript?

**Comment by ajzafar** [comment]
This is something I've wanted to implement for a while

**Comment by ajzafar** [comment]
I thought the `v:completed_item` would be needed for this before I could implement it. Now that it's in Vim 8, I went ahead and started working on it. Turns out I don't. Anyway, this is available in the `completion` branch. Instead of specifying `${1:foobar}` to denote `foobar` should be the placeholder, use `${1|foo|bar}` to get an insert mode completion menu with foo and bar as options. I think I'm going to call these selection stops. There are a couple problems I still need to fix:
1. Mirrors and substitutions and such are unavailable inside of selection items, so `${2|foo|$1|bar}` to make the value of the first stop an option does not work. Just the literal text `$1` will show up.
2. If a selection is made and then modified, SnipMate doesn't notice. That means if you come back to the selection stop, your modifications will be lost in favor of one of the original options.
